scripts/release/set-version.sh

```diff
@@ -28,61 +26,25 @@ do
     esac
 done
 
-function replace {
-    local file="$1"
-    local region="replace-version"
-
-    if (( $# == 2 )); then
-        region="$region $2"
-    fi
-
-    comment_begin="(#|\/\/|<!--)"
-    comment_end="( -->)?$"
-
-    suffix='(-[0-9a-zA-Z.]+)?'
-    num='[0-9]+'
-    # Replace both the version itself, and the sliding tags
-    #
-    # There is a caveat here for the major version. It is rather ambiguous, because
-    # it is just a single number, there are no dots in it that could identify it as
-    # semver version. So for the major version we require that it is always specified
-    # with the `v` prefix e.g. `v1`.
-    with_log sed --regexp-extended --in-place --file - "$file" <<EOF
-        /$comment_begin region $region$comment_end/,/$comment_begin endregion $region$comment_end/ \
-        {
-            s/(v|\W)$num\.$num\.$num$suffix/\1$new_version/g
-            s/(v|\W)$num\.$num$suffix/\1$new_major_minor/g
-            s/v$num$suffix/v$new_major/g
-        }
-EOF
-}
+# Set the rust toolchain in the installation scripts and other docs to the version
+# that was used in the development of the current release.
+rust_toolchain=$(sed -n 's/^channel\s*=\s*"nightly-\(.*\)"/\1/p' rust-toolchain.toml)
 
-files=($(\
-    find . -type f \
-        \( -name "*.rs" \
-        -o -name "*.md" \
-        -o -name "*.toml" \
-        -o -name "*.sh" \
-        -o -name "*.ps1" \
-        \)\
-))
+replace_date_in_regions "rust toolchain release" "$rust_toolchain"
```

Comment:
Further down, there is an if statement which conditionally replaces the version for `dev` or `stable`. Should we maybe to the same separation here? I imagine, we only want to do this replacement, when a stable version is released :thinking:

Reply:
Yeah, an unstable version release will be a special-case. In that case, we shouldn't update the toolchain versions in the docs, but we should update it in the installation scripts, and then return back the previous version in the installation scripts when the next dev version commit is made.
I don't like where it goes...

---

I'm thinking that what if we have a mutable `master` pre-release that we update every time a new commit is merged to master. For example, like it is done [here](https://github.com/bjorn3/rustc_codegen_cranelift/releases/tag/dev). 
Then the `master` installation script would install the true latest dev version of marker pre-compiled binaries, and it can use the latest toolchain version we have in the repo. 

For example, if someone reported a bug, and we fixed it in master, we closed the issue, Then  we can ask the user who reported the issue to check if it works for them by installing the master build of marker binaries and using a git dependency in their `marker_api`, and if the problem still exists they can reopen the issue after testing.

Reply:
I created an issue https://github.com/rust-marker/marker/issues/286.
But for now, I'll try to think of a crutch here..

Reply:
The crutch that I'm thinking off would require an additional git commit to revert the change in the toolchain version in the installation script. That by itself would require some effort to implement that I'd rather spend on https://github.com/rust-marker/marker/issues/286. Let's say that for now doing a pre-release will suffer from the problem that the toolchain version in master for installation scripts is updated to the current one. It's easy to fix it manually be returning back the changes to the toolchain version made by the release process in master. I just assume that we'll do a pre-release soon.